src/gluonts/model/rotbaum/_predictor.py

```diff
@@ -0,0 +1,175 @@
+        self.start_date = start_date
+        self.freq = freq
+        self.prediction_length = prediction_length
+        self.item_id = None
+
+    @validated()
+    def quantile(self, q: float) -> np.array:
+        """
+        Returns np.array, where the i^th entry is the estimate of the q
+        quantile of the conditional distribution of the value of the i^th
+        step in the forecast horizon.
+        """
+        assert 0 <= q <= 1
+        return np.array(
+            list(
+                chain(
+                    *[
+                        model.predict(self.featurized_data, q)
+                        for model in self.models
+                    ]
+                )
+            )
+        )
+
+    def estimate_dists(self) -> np.array:
+        """
+        Returns np.array, where the i^th entry is an estimated sampling from
+        the conditional distribution of the value of the i^th step in the
+        forecast horizon.
+        """
+        return np.array(
+            list(
+                chain(
+                    *[
+                        model.estimate_dist(self.featurized_data)
+                        for model in self.models
+                    ]
+                )
+            )
+        )
+
+
+class TreePredictor(RepresentablePredictor):
+    """
+    A predictor that uses a QRX model for each of the steps in the forecast
+    horizon. (In other words, there's a total of prediction_length many
+    models being trained. In particular, this predictor does not learn a
+    multivariate distribution.) The list of these models is saved under
+    self.model_list.
+    """
+
+    @validated()
+    def __init__(
+        self,
+        context_length: int,
+        prediction_length: int,
+        n_ignore_last: int = 0,
+        model_params=None,
+        freq=None,
+    ) -> None:
+
+        self.preprocess_object = PreprocessOnlyLagFeatures(
+            context_length,
+            forecast_horizon=prediction_length,
+            stratify_targets=False,
+            n_ignore_last=n_ignore_last,
+        )
+        self.context_length = context_length
+        self.model_params = model_params
+        self.prediction_length = prediction_length
+        self.freq = freq
+        self.model_list = None
+
+    @validated()
+    def __call__(self, training_data: Dataset):
+        assert training_data
+        if self.freq is not None:
+            assert self.freq == next(iter(training_data))["start"].freq
+        else:
+            self.freq = next(iter(training_data))["start"].freq
+        self.preprocess_object.preprocess_from_list(
+            ts_list=training_data, change_internal_variables=True
+        )
+        feature_data, target_data = (
+            self.preprocess_object.feature_data,
+            self.preprocess_object.target_data,
+        )
+        n_models = self.prediction_length
+        print(f"Length of forecast horizon: {n_models}")
```

Comment:
We shouldn't have `print` statements. You can use the `logging` module.

Reply:
Changed to logging.